lib/exercism/user_finished_tracks.rb

```diff
@@ -0,0 +1,63 @@
+require_relative '../../x'
+class UserFinishedTracks
+  attr_reader :user
+  def initialize(user)
+    @user = user
+  end
+
+  # Return Hahs with, finished tracks
+  def list
+    @list ||= completed_tracks
+  end
+
+  def any?
+    !list.empty?
+  end
+
+  private
+
+  def completed_tracks
+    decorated_db_info.select do |hash|
+      hash['completed_count'].to_i == hash['total_in_track']
+    end
+  end
+
+  # Append more information, to the data extracted from the db.
+  # Example:
+  # [
+  #   {
+  #     'track_id' => 'go', 'completed_exercise' => 12,
+  #     'problems_count' => 30, 'language' => 'Go'
+  #   },
+  #   { 'track_id' => 'ruby', 'completed_exercise' => 60,
+  #     'problems_count' => 70, 'language' => 'Ruby'
+  #   }
+  # ]
+  def decorated_db_info
+    completed_exercises_by_language_and_user.each do |hash|
+      hash['total_in_track'] = track_problems_count[hash['track_id']]['problems_count']
+      hash['language'] = track_problems_count[hash['track_id']]['language']
+      hash
+    end
+  end
+
+  def completed_exercises_by_language_and_user
```

Comment:
Everything in this class is about the user, so we shouldn't have to say `_and_user` in the name.

Also, these aren't the completed exercises, these are the _started tracks_.